src/missed_spans.rs

```diff
@@ -143,11 +143,12 @@ impl<'a> FmtVisitor<'a> {
                     line_start = offset + subslice.len();
 
                     if let Some('/') = subslice.chars().skip(1).next() {
+                        // Add a newline after line comments
                         self.buffer.push_str("\n");
                     } else if line_start < snippet.len() {
-                        let x = (&snippet[line_start..]).chars().next().unwrap() != '\n';
-
-                        if x {
+                        // For other comments add a newline if there isn't one at the end already
+                        let c = snippet[line_start..].chars().next().unwrap();
+                        if c != '\n' && c != '\r' {
                             self.buffer.push_str("\n");
```

Comment:
Currently it seems that `\n` is always used as newlines and it is only when writing to file/stdout etc that the actual line ending is decided on. https://github.com/rust-lang-nursery/rustfmt/blob/master/src/filemap.rs#L75-L80

It might be a good idea to always use the correct line but that should probably be a separate PR (?)